workspaces/liferay-testray-workspace/client-extensions/liferay-testray-custom-element/src/components/ManagementToolbar/ManagementToolbarFilter.tsx

```diff
@@ -269,68 +243,69 @@ const FilterBody: React.FC<FilterBodyProps> = ({
 				>
 					{i18n.translate('clear')}
 				</ClayButton>
-			</div>
+			</ClayDropDown.Section>
 		</div>
 	);
 };
 
-const MENU_POPOVER_HEIGHT = 580;
-
 const ManagementToolbarFilter: React.FC<ManagementToolbarFilterProps> = ({
 	applyFilters = true,
 	fieldOptions = {},
 	filterSchema,
 	isLoading = false,
 }) => {
+	const buttonRef = useRef<HTMLButtonElement | null>(null);
+
 	const [visible, setVisible] = useState(false);
-	const ref = useRef<HTMLButtonElement>(null);
 
-	const [position, setPosition] = useState<number>(MENU_POPOVER_HEIGHT);
+	const handleExpand = (
+		event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+	) => {
+		buttonRef.current = event.target as HTMLButtonElement;
 
-	const popoverAlignPosition =
-		position < MENU_POPOVER_HEIGHT ? 'top-right' : 'bottom-right';
+		setVisible(!visible);
+	};
 
 	return (
-		<ClayPopover
-			alignPosition={popoverAlignPosition}
-			className={classNames('popover-management-toolbar', {
-				'popover-management-toolbar-small':
-					filterSchema?.fields?.length === 1,
-			})}
-			closeOnClickOutside
-			disableScroll
-			onShowChange={setVisible}
-			show={visible && position > 0}
-			trigger={
-				<ClayButton
-					className="management-toolbar-buttons nav-link"
-					displayType="unstyled"
-					ref={ref}
-				>
-					<span className="navbar-breakpoint-down-d-none">
-						<ClayIcon
-							className="inline-item inline-item-after inline-item-before"
-							symbol="filter"
-						/>
-					</span>
-
-					<span className="navbar-breakpoint-d-none">
-						<ClayIcon symbol="filter" />
-					</span>
-				</ClayButton>
-			}
-		>
-			<FilterBody
-				applyFilters={applyFilters}
-				buttonRef={ref}
-				fieldOptions={fieldOptions}
-				filterSchema={filterSchema}
-				isLoading={isLoading}
-				setPosition={setPosition}
-				setVisible={setVisible}
-				visible={visible}
-			/>
-		</ClayPopover>
+		<>
+			<ClayButton
+				className="management-toolbar-buttons nav-link"
+				displayType="unstyled"
+				onClick={handleExpand}
+			>
+				<span className="navbar-breakpoint-down-d-none">
+					<ClayIcon
+						className="inline-item inline-item-after inline-item-before"
+						symbol="filter"
+					/>
+				</span>
+
+				<span className="navbar-breakpoint-d-none">
+					<ClayIcon symbol="filter" />
+				</span>
+			</ClayButton>
+
+			<ClayDropDown.Menu
+				active={visible}
+				alignElementRef={buttonRef}
+				alignmentPosition={3}
+				className={classNames('dropdown-management-toolbar', {
+					'dropdown-management-toolbar-small':
+						filterSchema?.fields?.length === 1,
```

Comment:
```suggestion
				className={classNames('dropdown-management-toolbar', {
					'dropdown-management-toolbar-small':
						hasOneFilter,
```
Can we save this value in a variable and apply here? So it would be better to read and understand. What do you think?